sdk/eventhub/event-hubs/src/batchingReceiver.ts

```diff
@@ -351,10 +351,10 @@ export class BatchingReceiver extends EventHubReceiver {
       });
 
     const jitterInSeconds = randomNumberFromInterval(1, 4);
-    const times =
-      retryOptions && retryOptions.retryCount && retryOptions.retryCount > 0
-        ? retryOptions.retryCount
-        : Constants.defaultRetryAttempts;
+    const maxRetries =
+      retryOptions && retryOptions.maxRetries && retryOptions.maxRetries > 0
```

Comment:
If user doesnt want to retry, then we expect them to pass 0 to `maxRetries` and here we don't respect that 0 and end up using the default value

Same in other places where we pass the value provided by the user.

I was also thinking... why not just pass `retryOptions && retryOptions.maxRetries` directly and have all the handling of 0, negative numbers, default values be done inside the `retry` function in core-amqp. This way we don't need to do the same everytime we call `retry`

Reply:
Agree, the handling of 0 case is a bug and affects other parameters too. We might want to revisit this in a different PR and look for, update all such instances. Updated ones pertaining to `maxRetries` here and the delayInSeconds too. But the delayInSeconds  value was being subsequently used for some other calculations, so that had to stay in Event Hubs.